frameworks/shared/caller.py

```diff
@@ -82,7 +82,8 @@ def run_in_venv(caller_file, script_file: str, *args,
                 input_data: Union[dict, ns], dataset: Dataset, config: TaskConfig,
                 options: Union[None, dict, ns] = None,
                 process_results=None,
-                python_exec=None):
+                python_exec=None,
+                retain_tmp_env: bool = True):
```

Comment:
I like the idea, especially for TMP properties, but I'd suggest to make it more generic then in case frameworks want to propagate some other variables:
```suggestion
                retained_env_vars: List[str] = ['TMP', 'TEMP', 'TMPDIR']):
```